app/lib/efile/ny/it201.rb

```diff
@@ -367,7 +384,56 @@ def filing_status_single?
       end
 
       def line_or_zero(line)
-        @computed[line] || 0
+        @lines[line.to_sym]&.value || 0
+      end
+
+      class TaxFormLine
+        attr_reader :line_id
+        attr_reader :value
+
+        def initialize(line_id, value, immutable: false)
+          @line_id = line_id
+          @value = value
+        end
+        
+        def value_or_zero
+          value || 0
+        end
+      end
+
+      class CalculatedTaxFormLine < TaxFormLine
+        attr_accessor :value
+
+        def source_description
+          method_name = "calculate_#{line_id.to_s.sub('AMT_', 'line_')}".to_sym
+          method = (Efile::Ny::It201.instance_method(method_name) rescue nil)
+          if method
+            method.source
+          else
+            "Unknown!"
+          end
+        end
+      end
+
+      class ImmutableTaxFormLine < TaxFormLine
+        def initialize(line_id, data_source, field_or_constant)
+          @line_id = line_id
+          @value = field_or_constant.is_a?(Symbol) ? data_source.send(field_or_constant) : field_or_constant
+          @data_source = data_source
+          @field_or_constant = field_or_constant
+        end
+
+        def source_description
+          "#{@data_source.class}##{@field_or_constant}"
+        end
+
+        def value=(new_value)
```

Comment:
Everything is sorta immutable due to the fact that we already know the value at construction time 🏗️